core/src/lib.rs

```diff
@@ -40,3 +40,20 @@ pub use self::from_variant::FromVariant;
 pub use quote::ToTokens;
 #[doc(hidden)]
 pub use syn;
+
+#[macro_export]
+macro_rules! parse_attr_into {
+    ($attr:ident as $ty:ty) => {
```

Comment:
I'd suggest making one branch with `as $ty:ty` and one without, as seen in [`syn::parse_macro_input`](https://docs.rs/syn/latest/syn/macro.parse_macro_input.html) - in the case where `as $ty` isn't provided, you can use `$crate::FromMeta::from_list` and expect Rust to figure out the type based on what's expected.